src/github.com/getlantern/balancer/balancer.go

```diff
@@ -102,11 +102,17 @@ func (b *Balancer) Dial(network, addr string) (net.Conn, error) {
 		trustedOnly = true
 	}
 
+	var lastDialer *dialer
 	for i := 0; i < dialAttempts; i++ {
 		d, pickErr := b.pickDialer(trustedOnly)
 		if pickErr != nil {
 			return nil, pickErr
 		}
+		if d == lastDialer {
```

Comment:
The only thing is that the dialer picking is weighted right? So it's quite likely it will pick the same dialer over and over again, likely using up all `dialAttempts` quite quickly. Is that correct? 

It's almost like as soon as we see a single failure we want to switch modes and dial several simultaneously or something like that.